app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarDetailsFragment.kt

```diff
@@ -66,20 +76,75 @@ class CalendarDetailsFragment : BottomSheetDialogFragment() {
 
         if (calendarItem.url.isEmpty()) {
             buttonsContainer.visibility = View.VISIBLE
-            deleteButton.setOnClickListener {
-                (activity as CalendarActivity).deleteEvent(calendarItem.nr)
-            }
-            editButton.setOnClickListener { (activity as CalendarActivity).editEvent(calendarItem)}
+            deleteButton.setOnClickListener { displayDeleteDialog(calendarItem.nr) }
+            editButton.setOnClickListener { listener.onEditEvent(calendarItem) }
         } else {
             buttonsContainer.visibility = View.GONE
         }
     }
 
+    private fun displayDeleteDialog(eventId: String) {
+        AlertDialog.Builder(context)
+                .setTitle(R.string.event_delete_title)
+                .setMessage(R.string.delete_event_info)
+                .setPositiveButton(R.string.delete) { _, _ -> deleteEvent(eventId) }
+                .setNegativeButton(R.string.cancel, null)
+                .show()
+    }
+
+    private fun deleteEvent(eventId: String) {
+        context?.let { c ->
```

Comment:
In those cases, I like a guarding elvis operator better.
```
val c = context ?: return
```

Reply:
You’re right, this is a little cleaner due to less indentation. Updated accordingly.